jupyterhub/templates/image-puller/job.yaml

```
@@ -36,7 +36,7 @@ spec:
             - /image-awaiter
             - -ca-path=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt
             - -auth-token-path=/var/run/secrets/kubernetes.io/serviceaccount/token
-            - -api-server-address=https://$(KUBERNETES_SERVICE_HOST):$(KUBERNETES_SERVICE_PORT)
+            - -api-server-address=https://kubernetes.default.svc
```

```suggestion
            - -api-server-address=https://kubernetes.default.svc:$(KUBERNETES_SERVICE_PORT)
```

As a precaution, lets not assume this service doesn't always use port 443.